pkg/dogstatsd/server.go

```diff
@@ -260,77 +251,107 @@ func nextMessage(packet *[]byte) (message []byte) {
 	return message
 }
 
-func (s *Server) parsePacket(packet *listeners.Packet, metricSamples []*metrics.MetricSample, events []*metrics.Event, serviceChecks []*metrics.ServiceCheck) ([]*metrics.MetricSample, []*metrics.Event, []*metrics.ServiceCheck) {
-	extraTags := s.extraTags
+func (s *Server) parsePackets(batcher *batcher, packets []*listeners.Packet) {
+	for _, packet := range packets {
+		originTags := findOriginTags(packet.Origin)
+		//log.Tracef("Dogstatsd receive: %s", packet.Contents)
+		for {
+			message := nextMessage(&packet.Contents)
+			if message == nil {
+				break
+			}
+			if s.Statistics != nil {
+				s.Statistics.StatEvent(1)
+			}
+			messageType := findMessageType(message)
 
-	log.Tracef("Dogstatsd receive: %s", packet.Contents)
```

Comment:
There is some overhead asociated (one alloc because `packet.Contents` gets converted to `interface{}` and escapes) but it was a mistake removing it, I'll be reverting